geode-core/src/main/java/org/apache/geode/internal/tcp/ConnectionTable.java

```diff
@@ -519,8 +519,12 @@ private void scheduleIdleTimeout(Connection conn) {
           if (!closed) {
             IdleConnTT task = new IdleConnTT(conn);
             conn.setIdleTimeoutTask(task);
-            getIdleConnTimer().scheduleAtFixedRate(task, owner.idleConnectionTimeout,
-                owner.idleConnectionTimeout);
+            synchronized (task) {
```

Comment:
does synchronizing here relate to removing the `continue`s in the timer code?

Reply:
The synchronizations and cancel checks fix GEODE-7884.  The SystemTimer changes, except for adding isCancelled(), are unrelated.  I made those changes so that I could test isCancelled() and add test coverage for SystemTimer.

Reply:
roger that